### PR TITLE
Reconcile SemanticModel behavior for imports/using aliases with errors across VB and C#.

### PR DESCRIPTION
**Pull request by AlekseyTs**
This change is related to issue #2805.

When an alias target cannot be resolved,
GetSymbolInfo(<alias reference node>) – returns Nothing in VB, but returns an ErrorTypeSymbol in C#
GetAliasInfo(<alias reference node>) – returns Nothing in VB, but returns corresponding AliasSymbol in C#.

GetSymbolInfo should never return error types, so VB behavior is correct and this change adjusts C# behavior accordingly.
GetAliasInfo should return an AliasSymbol even if the alias target cannot be resolved, so C# behavior is correct. VB behavior was caused by the fact that, if target couldn’t be resolved, the AliasSymbol wasn’t even created. This change ensures that the AliasSymbols are created.

@VSadov, @gafter, @agocke, @jaredpar, @khyperia Please review.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
Ping. @VSadov, @gafter, @agocke, @jaredpar, @khyperia Please review. Note, the CI build failed due to Mac/Linux issues, I'll disable new C# test on these platforms and open an issue to track the failure.

**Comment by AlekseyTs** [comment]
@AnthonyDGreen FYI

**Comment by AnthonyDGreen** [comment]
This logic looks sound to me. I approve.

**Comment by VSadov** [comment]
LGTM

File: src/Compilers/VisualBasic/Portable/Binding/Binder_Imports.vb [hunk 1]
```diff
@@ -90,56 +90,53 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
...
-                                                                 aliasTarget,
-                                                                 If(binder.BindingLocation = BindingLocation.ProjectImportsDeclaration, NoLocation.Singleton, aliasIdentifier.GetLocation()))
 
-                                data.AddAlias(binder.GetSyntaxReference(aliasImportSyntax), aliasText, aliasSymbol, aliasImportSyntax.SpanStart)
+                            If ReportUseSiteErrorForAlias(useSiteErrorInfo) Then
+                                Binder.ReportDiagnostic(diagBag, aliasImportSyntax, useSiteErrorInfo)
                             End If
                         End If
+
+                        Dim aliasSymbol = New AliasSymbol(binder.Compilation,
```
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Binding/Binder_Imports.vb]: I am surprised to see us creating a new alias symbol here. Isn't there one already in existence?

File: src/Compilers/VisualBasic/Portable/Binding/Binder_Imports.vb [hunk 2]
```diff
@@ -90,56 +90,53 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
...
-            ''' Checks use site error and returns True in case the alias should still be created for the
-            ''' type with this site error. In current implementation checks for errors ##36924, 36925
+            ''' Checks use site error and returns True in case it should be reported for the alias. 
+            ''' In current implementation checks for errors #36924 and #36925
             ''' </summary>
-            Private Shared Function AllowAliasWithUseSiteError(useSiteErrorInfo As DiagnosticInfo) As Boolean
-                Return useSiteErrorInfo Is Nothing OrElse
-                        useSiteErrorInfo.Code = ERRID.ERR_CannotUseGenericTypeAcrossAssemblyBoundaries OrElse
-                        useSiteErrorInfo.Code = ERRID.ERR_CannotUseGenericBaseTypeAcrossAssemblyBoundaries
+            Private Shared Function ReportUseSiteErrorForAlias(useSiteErrorInfo As DiagnosticInfo) As Boolean
```
Comment (gafter) [inline src/Compilers/VisualBasic/Portable/Binding/Binder_Imports.vb]: Name slightly confusing, as this doesn't report any errors.

**Comment by gafter** [comment]
LGTM, though consider the inline comments.

File: src/Compilers/VisualBasic/Portable/Binding/Binder_Imports.vb (hunk 1 above)
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Binding/Binder_Imports.vb]: No. BTW this is an old code, it just got shifted to the left.

File: src/Compilers/VisualBasic/Portable/Binding/Binder_Imports.vb (hunk 2 above)
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Binding/Binder_Imports.vb]: I'll use suggested name for the function.

**Comment by AlekseyTs** [comment]
@dotnet-bot Test this please

**Pull request merged**